### PR TITLE
SCP 2708 - ACTUS ANN contracts

### PR DESCRIPTION
**Pull request by yveshauser**
Pre-submit checklist:
- Branch
    - [x] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [x] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

### CODE REVIEW

**Review by palas — APPROVED** [review]
Looks good to me, I just wrote a couple of comments.

Refactorings did seem to simplify things in general, even though I keep feeling that `Maybe` handling is producing too much boilerplate.

And thanks for the `README.md`, really helpful and concise.

Do wait for the review from @d-velev because my understanding of how it actually work is very limited.

File: marlowe-actus/README.md
Comment (yveshauser) [reply marlowe-actus/README.md:9]: Yes, agree. I'm currently adding illustrative examples for all contract types to the README for the next sprint. In general I'd like to put more details, in order for the reader to better understand what these contract are intended to do.

**Comment by yveshauser** [comment]
> Refactorings did seem to simplify things in general, even though I keep feeling that `Maybe` handling is producing too much boilerplate.

Yes, `Maybe` handling is annoying. Pattern matching directly in the record syntax is probably the best approach to this. I'd like to have this pattern in all places - currently for example `ContractScheduleModel.hs` is not yet done. I will change it too, as soon as we have merged in the LAX contract type. 

And I still hope to get rid off the different entry points into the model, in order not to have the dispatching duplicated (`payoff` and `payoffFs`, resp.  `stateTransition` and `stateTransitionFs`)

**Review by d-velev — APPROVED** [review]
Sorry about the delayed review, looks good

**Pull request merged**
